pss/pss.go

```diff
@@ -52,11 +61,27 @@ func (p *Pss) Send(ctx context.Context, targets [][]byte, topic trojan.Topic, pa
 		return nil, err
 	}
 
+	tc.WithTagID(p.tag.Uid)
+
 	// SAVE trojanChunk to localstore, if it exists do nothing as it's already peristed
-	// TODO: for second phase, use tags --> listen for response of recipient, recipient offline
 	if _, err = p.localStore.Put(ctx, chunk.ModePutUpload, tc); err != nil {
 		return nil, err
 	}
+	p.tag.Inc(chunk.StateStored)
```

Comment:
tag creation and handling should be local to this function
a monitor objrct should be returned , the user ecperience being something like

```golang
monitor, err := pss.Send(...)
for state := range monitor.states {
    log.Log("message has been "%s"
}
```

Reply:
ok, that makes more sense.
I will refactor to this.